R/zchunk_LB112.ag_prodchange_R_C_Y_GLU.R

```diff
@@ -40,62 +41,292 @@ module_aglu_LB112.ag_prodchange_R_C_Y_GLU_DISABLED <- function(command, ...) {
     L100.LDS_ag_HA_ha <- get_data(all_data, "L100.LDS_ag_HA_ha")
     L103.ag_Prod_Mt_R_C_Y_GLU <- get_data(all_data, "L103.ag_Prod_Mt_R_C_Y_GLU")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # Initial preparation of CROSIT database
+    # Mapping file for CROSIT country name and country ID
+    AGLU_ctry %>%
+      select(CROSIT_ctry, CROSIT_country_ID) %>%
+      unique() %>%
+      na.omit() -> ctry_id
+
+    # Mapping file for CROSIT crop name and crop ID
+    FAO_ag_items_PRODSTAT %>%
+      select(CROSIT_crop, CROSIT_cropID) %>%
+      unique() %>%
+      na.omit() -> crop_id
+
+    # Build a table with all specified ag productivity years, prepare for interpolation
+    FAO_ag_CROSIT %>%
+      left_join_error_no_match(ctry_id, by = c("country_ID" = "CROSIT_country_ID")) %>%
+      left_join_error_no_match(crop_id, by = c("crop_ID" = "CROSIT_cropID")) %>%
+      select(CROSIT_ctry, CROSIT_crop) %>%
+      unique() %>%
+      repeat_add_columns(tibble(SPEC_AG_PROD_YEARS)) -> interp.tbl
+
+    # Intial CROSIT data clean-up and interpolation of yields
+    FAO_ag_CROSIT %>%
+      # Replace country IDs with names
+      left_join_error_no_match(ctry_id, by = c("country_ID" = "CROSIT_country_ID")) %>%
+      # Replace crop IDs with names
+      left_join_error_no_match(crop_id, by = c("crop_ID" = "CROSIT_cropID")) %>%
+      # NOTE: Only use total production and harvested area from CROSIT (not separating rainfed and irrigated).
+      select(CROSIT_ctry, CROSIT_crop, year, HA_kha, Yield_kgHa, Prod_kt,
+             -HA_kha_rainfed, -Yield_kgHa_rainfed, -Prod_kt_rainfed, -HA_kha_irrigated, -Yield_kgHa_irrigated, -Prod_kt_irrigated) %>%
+      # Convert the needed numeric variables from integer to double for calculation
+      mutate(HA_kha = as.double(HA_kha),
+             Yield_kgHa = as.double(Yield_kgHa),
+             Prod_kt = as.double(Prod_kt)) %>%
+      # Some regions have 0 production with positive harvested area and a reported yield.
```

Comment:
Starting here in this pipe, there are several adjustments made of production and harvest area data. However, it seems like these two variables are not used for anything else in this chunk, and removing these lines does not affect any outputs.